client/question/domain/CodeEvalResultObjectFactory.js

```diff
@@ -174,6 +189,39 @@ tie.factory('CodeEvalResultObjectFactory', [
       return null;
     };
 
+  /**
+     * Compares the results to the expected values of the test cases, and
+     * returns the test number of the first failed test case (0-indexed). The
+     * test number is independent of tasks and testSuites but rather the overall
+     * test case number.
+     *
+     * @param {Array<Task>} tasks The list of tasks for the current question.
+     * @returns {number} The index of the first failed test case, or index of
+     * last test case if all test cases passed.
+     */
+    CodeEvalResult.prototype.getIndexOfFirstFailedTest = function(tasks) {
+      if (this._observedOutputs.length === 0) {
+        return 0;
+      }
+
+      var userOutput = this._observedOutputs;
```

Comment:
Why rename, especially since you use the original variable in line 214?

Also consider `observedOutputs` instead, it's a bit confusing to call the same thing "output" and "outputs".

Reply:
Reverted to original name.